tools/duckduckgo/ddg.go

```diff
@@ -0,0 +1,42 @@
+package duckduckgo
+
+import (
+	"context"
+	"errors"
+
+	"github.com/tmc/langchaingo/tools"
+	"github.com/tmc/langchaingo/tools/duckduckgo/internal"
+)
+
+type Tool struct {
+	client *internal.Client
+}
+
+var _ tools.Tool = Tool{}
+
+func New() (*Tool, error) {
+	return &Tool{}, nil
+}
+
+func (t Tool) Name() string {
+	return "DuckDuckGo Search"
+}
+
+func (t Tool) Description() string {
+	return `
+	"A wrapper around DuckDuckGo Search."
+	"Free search alternative to google and serpapi."
+	"Input should be a search query."`
+}
+
+func (t Tool) Call(_ context.Context, input string) (string, error) {
```

Comment:
the context shouldn't be ignored and should be used in the underlying search.

Reply:
Rewrote the ddg API request to include context, should be ok now...